extension-spring/src/main/java/io/smallrye/openapi/spring/SpringAnnotationScanner.java

```diff
@@ -67,7 +68,7 @@ public boolean isPostMethod(final MethodInfo method) {
 
     @Override
     public boolean isScannerInternalResponse(Type returnType) {
-        return returnType.name().equals(SpringConstants.RESPONSE_ENTITY);
+        return returnType.name().equals(SpringConstants.RESPONSE_ENTITY) && returnType.kind() != Kind.PARAMETERIZED_TYPE;
```

Comment:
I think you might want to check for an instance of `HttpEntity` where the type argument is a wildcard, e.g. `HttpEntity<?>`. `ResponseEntity` extends `HttpEntity` so you will cover a few more scenarios.

See `OpenApiDataObjectScanner`'s static init block for an example of how to get the `ClassInfo` of types that cannot be assumed to be in the application's `IndexView` and also how to check if a type inherits from another using `TypeUtil#isA`.

Related to the type of response, you may also look at creating an implementation of `AnnotationScannerExtension` that is used in the Spring scanner with the `resolveAsyncType` overridden. You would want logic similar to that in `SchemaFactory#resolveAsyncType` to pull the type argument from the `HttpEntity<T>` or `ResponseEntity<T>`. The overridden method would then be called from `SchemaFactory#resolveAsyncType`.

Reply:
I am not sure I understand what that "isScannerInternalResponse" is even supposed to do for Spring. JAX-RS Response is truly useless when it comes to figuring out response types. I'm tempted to always return false, to be honest.

Reply:
You're right, but if the parameter is a wildcard, that's no better than the JAX-RS `Response`.

Reply:
Yea I think for now return false, if we later figure out there is some class that is like Response then we update it.